reporters/GlobalSummaryReporter.js

```diff
@@ -489,183 +670,111 @@ class GlobalSummaryReporter {
     } else {
-      for (let event of this.finalFeeEvents) {
-        finalFeesPaid = finalFeesPaid.add(this.toBN(event.amount));
-        if (event.blockNumber >= this.startBlockNumberForPeriod && event.blockNumber < this.endBlockNumberForPeriod) {
-          finalFeesPaidPeriod = finalFeesPaidPeriod.add(this.toBN(event.amount));
-        }
-        if (
-          event.blockNumber >= this.startBlockNumberForPreviousPeriod &&
-          event.blockNumber < this.startBlockNumberForPeriod
-        ) {
-          finalFeesPaidPrevPeriod = finalFeesPaidPrevPeriod.add(this.toBN(event.amount));
-        }
-      }
-    }
-
-    if (Object.keys(allDvmStatsTable).length !== 0) {
+      const finalFeeData = await this._filterFinalFeeData(periods, this.finalFeeEvents);
       allDvmStatsTable = {
         ["final fees paid to store"]: {
-          cumulative: this.formatDecimalString(finalFeesPaid),
-          [this.periodLabelInHours]: this.formatDecimalString(finalFeesPaidPeriod),
-          ["Δ from prev. period"]: this.formatDecimalStringWithSign(finalFeesPaidPeriod.sub(finalFeesPaidPrevPeriod))
-        },
-        ["ongoing regular fees paid to store"]: {
-          cumulative: this.formatDecimalString(regularFeesPaid),
-          [this.periodLabelInHours]: this.formatDecimalString(regularFeesPaidPeriod),
+          cumulative: this.formatDecimalString(regFeeData.allFinalFeesPaid),
+          [this.periodLabelInHours]: this.formatDecimalString(regFeeData.periodFinalFeesPaid["period"]),
           ["Δ from prev. period"]: this.formatDecimalStringWithSign(
-            regularFeesPaidPeriod.sub(regularFeesPaidPrevPeriod)
+            regFeeData.periodFinalFeesPaid["period"].sub(regFeeData.periodFinalFeesPaid["prevPeriod"])
           )
-        },
-        ["ongoing late fees paid to store"]: {
-          cumulative: this.formatDecimalString(lateFeesPaid),
-          [this.periodLabelInHours]: this.formatDecimalString(lateFeesPaidPeriod),
-          ["Δ from prev. period"]: this.formatDecimalStringWithSign(lateFeesPaidPeriod.sub(lateFeesPaidPrevPeriod))
         }
       };
+    }
 
+    if (Object.keys(allDvmStatsTable).length !== 0) {
       console.table(allDvmStatsTable);
     }
   }
 
-  async _getLookbackTimeInBlocks(lookbackTimeInSeconds) {
-    const blockTimeInSeconds = await averageBlockTimeSeconds();
-    const blocksToLookBack = Math.ceil(lookbackTimeInSeconds / blockTimeInSeconds);
-    return blocksToLookBack;
-  }
-
-  async _constructTokenHolderList() {
-    const cumulativeTokenHolders = {};
+  /** *******************************************************
+   *
+   * Misc. helper methods
+   *
+   * *******************************************************/
+
+  // Returns token holder statistics since synthetic token inception and for the periods input.
+  // Statistics include:
+  // - count of unique token holders during a period
+  // - final account balance at the end of the period
+  async _constructTokenHolderList(periods) {
```

Comment:
This method is powerful and will likely be used outside of this file in the future. It parses through `Transfer` events and constructs a token holder account list, both cumulatively and for specific periods.